### PR TITLE
Added Enhanced Perk Filter

### PR DESCRIPTION
**Pull request by brootroot**
Issue: https://github.com/DestinyItemManager/DIM/issues/8387

Added a range filter for enhanced perks.

### CODE REVIEW

File: src/app/search/search-filters/sockets.tsx [hunk 1]
```diff
@@ -245,6 +246,27 @@ const socketFilters: FilterDefinition[] = [
...
+    destinyVersion: 2,
+    filter: ({ filterValue }) => {
+      const compare = rangeStringToComparator(filterValue);
+      return (item: DimItem) => {
+        let enhancedPerkCount = 0;
+        if (item.sockets) {
+          // count enhanced perks
+          for (const socket of item.sockets.allSockets) {
+            if (socket.plugged) {
+              isEnhancedPerk(socket.plugged) && enhancedPerkCount++;
```
Comment (bhollis) [inline src/app/search/search-filters/sockets.tsx:262]: You already had a conditional:

```suggestion
            if (socket.plugged && isEnhancedPerk(socket.plugged)) {
              enhancedPerkCount++;
```

File: src/app/search/search-filters/sockets.tsx [hunk 2]
```diff
@@ -245,6 +246,27 @@ const socketFilters: FilterDefinition[] = [
...
+            }
+          }
+        }
+        return compare(enhancedPerkCount ?? 0);
```
Comment (bhollis) [inline src/app/search/search-filters/sockets.tsx:266]: `enhancedPerkCount` can't ever be `undefined` or `null`.

```suggestion
        return compare(enhancedPerkCount);
```

**Review by bhollis — APPROVED** [review]

**Comment by robojumper** [comment]
This weapon matches `enhancedperk:>=1` because the shader has the Common rarity, which is what enhanced perks use.

grafik

should probably filter to weapon perks first before looking at plug rarity. @brootroot do you want to follow up with a fix?

**Comment by brootroot** [comment]
@robojumper I'll work on something tonight

**Pull request merged**